ios/DomainLayer/SharedDomain/Package.swift

```diff
@@ -20,7 +20,8 @@ let package = Package(
     dependencies: [
         // Dependencies declare other packages that this package depends on.
         // .package(url: /* package url */, from: "1.0.0"),
-        .package(url: "https://github.com/hmlongco/Resolver.git", .upToNextMajor(from: "1.0.0"))
+        .package(url: "https://github.com/hmlongco/Resolver.git", .upToNextMajor(from: "1.0.0")),
+        .package(url: "https://github.com/groue/Semaphore", .upToNextMinor(from: "0.0.6"))
```

Comment:
You are using it in the UIToolkit, so why it is added as a dependency of SharedDomain? 🤔 And as it is just one file maybe we can just copy-paste it into our repo (I would suggest to have it in Utilities package) so we can easily adjust the code if needed in the future?

Reply:
I've added it to SharedDomain because it's not an UI library. Will copy&paste 👍

Reply:
Done